wandb/core.py

```diff
@@ -75,8 +74,16 @@ def termlog(string='', newline=True, repeat=True):
         line = ''
     if not repeat and line in PRINTED_MESSAGES:
         return
-    PRINTED_MESSAGES.add(line)
-    click.echo(line, file=sys.stderr, nl=newline)
+    # Repeated line tracking limited to 1k messages
+    if len(PRINTED_MESSAGES) < 1000:
+        PRINTED_MESSAGES.add(line)
+    if os.getenv(env.SILENT):
+        from wandb import util
+        util.mkdir_exists_ok(os.path.dirname(util.get_log_file_path()))
```

Comment:
This stuff doesn't normally go to the log file, right? Just making sure this is intentional.

Reply:
Yeah, the idea is some users don't want any of our output in the terminal, so I just write it to the debug log in that case.